osbuild/mounts.py

```diff
@@ -36,6 +37,8 @@ def calc_id(self):
         m.update(json.dumps(self.info.name, sort_keys=True).encode())
         if self.device:
             m.update(json.dumps(self.device.id, sort_keys=True).encode())
+        if self.partition:
+            m.update(json.dumps(self.partition, sort_keys=True).encode())
         if self.target:
             m.update(json.dumps(self.target, sort_keys=True).encode())
```

Comment:
I started to make this change when I added `partition` to the init signature, but ended up not doing it because regardless the `target` will be unique for each mount anyway, so there was no way it wouldn't be unique.

I'm happy to have it added here, though. Just wondering if my logic ^^ was somehow flawed?

Reply:
Yeah, it's a bit overdoing it maybe, it's possible to mount things over /target multiple times (but of course highly unlikely in practice) that I felt it should be part of this (also because it just felt more "right" that the id changes if things change)